src/test/java/com/morgan/design/properties/conversion/DefaultPropertyConversionServiceUnitTest.java

```diff
@@ -52,11 +55,18 @@ public void shouldConvertBooleanValue() throws NoSuchFieldException, SecurityExc
 		assertThat((Boolean) convertPropertyForField("booleanValue", "true"), is(true));
 	}
 
+	@Test
+	public void shouldConvertDateValue() throws NoSuchFieldException, SecurityException, ParseException {
+		Date expected = new SimpleDateFormat("dd-MM-yyyy").parse("9-4-2017");
+		assertThat((Date) convertPropertyForField("dateValue", "9-4-2017"), is(expected));
+	}
+	
 	static class TestObject {
 		Period period = new Period();
 		LocalTime localTime = new LocalTime();
 		LocalDate localDate = new LocalDate();
 		LocalDateTime localDateTime = new LocalDateTime();
+		Date dateValue = new Date();
 		boolean booleanValue;
 	}
```

Comment:
Looks good - can you enhance the integration test `ReloadablePropertyPostProcessorIntTest` just to prove it all wires together correct and properties can be injected. Thanks for the contribution.